content/booklets/adacore-technologies-for-airborne-software/analysis.rst

```diff
@@ -0,0 +1,3180 @@
+would then be expressed differently, for example with the Global and Depends referring to the abstract state of the package rather
+than visible variables.
+
+Some low-level requirements might not be expressible using the aspect mechanism (for example timing constraints). A convenient
+approach during architecture definition is to separately specify those components whose requirements can be defined using contracts,
+from those that cannot.
+
+.. index:: single: Use Case 1a: Traditional development process excluding OOT; Implementation of hardware / software interfaces
+
+.. _Implementation_of_Hardware_Software_Interfaces:
+
+Implementation of Hardware / Software Interfaces
+''''''''''''''''''''''''''''''''''''''''''''''''
+Ada's type system makes it straightforward to implement hardware/software interfaces, while also detecting target incompatibilities
+at compile time. Such interfaces may be defined as part of the coding process, but performing this activity during the design process
+has a number of benefits. It may avoid duplication of effort and also helps prevent errors from being introduced during the
+translation from design to code. It also allows early error detection through compilation checks.
+
+.. index:: single: Ada language; package Interfaces
+
+**Package Interfaces**
+
+Applications sometimes need to use types that correspond exactly to the native numeric data representations supported on the target
+machine, for example 16- or 32-bit signed and unsigned integers. Such types are defined in package `Interfaces`, which is part of the
+standard Ada library. The exact set of types depends on the target but typically includes integer types such as `Unsigned_16`,
+`Unsigned_32`, `Integer_16`, and `Integer_32`, as well as several floating-point types. The unsigned integer types are especially useful
+for hardware / software interfacing since they support bitwise operations including shift and rotate functions.
+
+.. index:: single: Ada language; Specifying data representation
+
+**Specifying data representation**
+
+Embedded systems often need to deal with external data having a specific representation, and Ada has a variety of features to help
+meet this requirement. For example, the following can be defined:
+
+* the values of the elements in an enumeration type,
+* the layout of a record (size and position of each field, possibly with fields overlaid), and
+* the address, size, and/or alignment of a data object.
+
+The compiler will check that the specified representation is consistent with the target hardware.
+For example, :numref:`fig3` shows the required layout (on a "little-endian" machine) for a data object consisting of an unsigned 16-bit
+integer (`Num`), a 4-bit enumeration value (`Urgency`) that is either `Low`, `Medium`, or `High`, with the respective values 2, 5, and 10), and
+a Boolean flag (F).
+
+.. _fig3:
```

Comment:
Have you checked whether these figures render well in the dark mode version of the website? I assume there are no issues, but it's good to test that.